src/debug.c

```diff
@@ -1639,6 +1793,396 @@ void DebugMenu_CalculateTimeOfDay(struct ScriptContext *ctx)
+            sDebugMenuListData->data[1] = -1;
+        }
+        ClearStdWindowAndFrame(gTasks[taskId].tWindowId, TRUE);
+        RemoveWindow(gTasks[taskId].tWindowId);
+        ClearStdWindowAndFrame(gTasks[taskId].tSubWindowId, TRUE);
+        RemoveWindow(gTasks[taskId].tSubWindowId);
+        DestroyListMenuTask(gTasks[taskId].tMenuTaskId, NULL, NULL);
+        DestroyTask(taskId);
+        sDebugMenuListData->listId = 2;
+        Debug_RemoveCallbackMenu();
+        Debug_ShowMenu(DebugTask_HandleMenuInput_General, sDebugMenu_Actions_Trainers);
+    }
+}
+
+static void DebugAction_Trainers_ChooseTrainer(u8 taskId, u32 selection)
+{
+    ClearStdWindowAndFrame(gTasks[taskId].tWindowId, TRUE);
+    RemoveWindow(gTasks[taskId].tWindowId);
+
+    HideMapNamePopUpWindow();
+    LoadMessageBoxAndBorderGfx();
+    u32 windowId = AddWindow(&sDebugMenuWindowTemplateExtra);
+    DrawStdWindowFrame(windowId, FALSE);
+
+    CopyWindowToVram(windowId, COPYWIN_FULL);
+
+    //Set task data
+    gTasks[taskId].func = DebugAction_ChooseTrainerID_Select;
+    gTasks[taskId].tSubWindowId = windowId;
+    gTasks[taskId].tDigit = 0;
+    gTasks[taskId].tSelection = (s32)selection;
+
+    switch(gTasks[taskId].tSelection)
+    {
+    case TRAINERS_DEBUG_SELECTION_TRAINER1:
+        gTasks[taskId].tInput = sDebugMenuListData->data[0];
+        break;
+    case TRAINERS_DEBUG_SELECTION_TRAINER2:
+        gTasks[taskId].tInput = sDebugMenuListData->data[2];
+        break;
+    case TRAINERS_DEBUG_SELECTION_PARTNER:
+        gTasks[taskId].tInput = sDebugMenuListData->data[4];
+        break;
+    }
+    gTasks[taskId].tInitial = gTasks[taskId].tInput;
+    // Display initial trainer
+    Debug_Display_TrainerID(gTasks[taskId].tInput, gTasks[taskId].tSelection, 0, windowId);
+}
+
+#undef tSelection
+#undef tInitial
+
+static void DebugAction_Trainers_SwitchDoublesFlag(u8 taskId)
+{
+    if (sDebugMenuListData->data[2] != TRAINER_NONE || sDebugMenuListData->data[4] != PARTNER_NONE)
+        return;
+    if (sDebugMenuListData->data[5])
+        sDebugMenuListData->data[5] = FALSE;
+    else
+        sDebugMenuListData->data[5] = TRUE;
+}
+
+static void DebugAction_Trainers_SetRematch(u8 taskId)
+{
+    s32 rematchId = sDebugMenuListData->data[1];
+    u32 i;
+
+    if (rematchId == -1)
+    {
+        FlagToggle(TRAINER_FLAGS_START + sDebugMenuListData->data[0]);
+        return;
+    }
+
+    for (i = 0; i < REMATCHES_COUNT; i++)
+    {
+        if (gRematchTable[rematchId].trainerIds[i] == 0)
+            break;
+        if (!HasTrainerBeenFought(gRematchTable[rematchId].trainerIds[i]))
+        {
+            FlagToggle(TRAINER_FLAGS_START + gRematchTable[rematchId].trainerIds[i]);
+            return;
+        }
+    }
+
+    for (i = 0; i < REMATCHES_COUNT; i++)
+    {
+        if (gRematchTable[rematchId].trainerIds[i] == 0)
+            break;
+        FlagToggle(TRAINER_FLAGS_START + gRematchTable[rematchId].trainerIds[i]);
+    }
```

Comment:
```suggestion

    if (rematchId == -1)
    {
        FlagToggle(TRAINER_FLAGS_START + sDebugMenuListData->data[0]);
        return;
    }

    for (u32 i = 0; i < REMATCHES_COUNT; i++)
    {
        if (gRematchTable[rematchId].trainerIds[i] == 0)
            break;

        if (!HasTrainerBeenFought(gRematchTable[rematchId].trainerIds[i]))
        {
            FlagToggle(TRAINER_FLAGS_START + gRematchTable[rematchId].trainerIds[i]);
            return;
        }
    }

    for (u32 i = 0; i < REMATCHES_COUNT; i++)
    {
        if (gRematchTable[rematchId].trainerIds[i] == 0)
            break;

        FlagToggle(TRAINER_FLAGS_START + gRematchTable[rematchId].trainerIds[i]);
    }
```
Loop iterator and formatting